test/Polly.Core.Tests/Polly.Core.Tests.csproj

```diff
@@ -1,16 +1,17 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>net8.0;net7.0;net6.0</TargetFrameworks>
+    <TargetFrameworks>net9.0;net8.0</TargetFrameworks>
     <TargetFrameworks Condition="$([MSBuild]::IsOSPlatform('Windows'))">$(TargetFrameworks);net481</TargetFrameworks>
     <ProjectType>Test</ProjectType>
     <Nullable>enable</Nullable>
-    <Threshold>100</Threshold>
     <NoWarn>$(NoWarn);SA1600;SA1204</NoWarn>
     <Include>[Polly.Core]*</Include>
     <SuppressTfmSupportBuildWarnings>true</SuppressTfmSupportBuildWarnings>
   </PropertyGroup>
 
   <ItemGroup>
+    <!-- HACK net9.0 compilation for the tests fails due to missing Task.Delay() extension for TimeProvider without this reference -->
+    <PackageReference Include="Microsoft.Bcl.TimeProvider" Condition="$([MSBuild]::IsTargetFrameworkCompatible($(TargetFramework), 'net9.0'))" />
```

Comment:
@martintmk Any idea what's going on with this? Without an explicit reference to this package, I can't get the tests to compile for `net9.0`.

The only change I can see that might explain it is [this](https://github.com/dotnet/extensions/commit/16b40667278d24a23af0c839f4c277499b3fee19#diff-11865412b98c8eebb41fa5cd1c55c154b9b58ca5e7dbe41bd651124bdc5e1b9f) from https://github.com/dotnet/extensions/pull/5058.